src/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -56,32 +56,39 @@
     private ValidationStringency validationStringency;
 
     /**
-     * Open CRAM data for reading using either the file or the input stream
-     * supplied in the arguments. The
-     * {@link htsjdk.samtools.Defaults#REFERENCE_FASTA default} reference fasta
-     * file will be used.
+     * Create a CRAMFileReader from either a file or input stream using the {@link ReferenceSource#getDefaultCRAMReferenceSource() default}
+     * reference source.
      *
      * @param cramFile CRAM file to open
-     * @param inputStream   CRAM stream to read
+     * @param inputStream CRAM stream to read
+     *
+     * @throws IllegalArgumentException if the {@code cramFile} and the {@code inputStream} are both null
+     * @throws IllegalStateException if a {@link ReferenceSource#getDefaultCRAMReferenceSource() default}
+     * reference source cannot be acquired
      */
     public CRAMFileReader(final File cramFile, final InputStream inputStream) {
-        this(cramFile, inputStream, new ReferenceSource(Defaults.REFERENCE_FASTA));
+        this(cramFile, inputStream, ReferenceSource.getDefaultCRAMReferenceSource());
```

Comment:
The docs for the called constructor (and several of the other inputStreams below) are wrong. I'll fix them as part of this PR.

And I totally agree this constructor is silly and confusing, not only because of the stream/file thing, but also because its the only one of the (7!) CRAMFileReader constructors that doesn't have a reference argument, so its only useful if you're using a default reference source. But its already public. Its only use in htsjdk is from SamFileReader, which is deprecated - should we deprecate it ?